File: java/BarangElektronik.java
import java.util.Scanner;

public class BarangElektronik {
    private int id;
    private String nama;
    private int harga;
    private int stok;

    public BarangElektronik() {
    }

    public BarangElektronik(int id, String nama, int harga, int stok) {
        this.id = id;
        this.nama = nama;
        this.harga = harga;
        this.stok = stok;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public void setNama(String nama) {
        this.nama = nama;
    }

    public String getNama() {
        return nama;
    }

    public void setHarga(int harga) {
        this.harga = harga;
    }

    public int getHarga() {
        return harga;
    }

    public void setStok(int stok) {
        this.stok = stok;
    }

    public int getStok() {
        return stok;
    }

    // CRUD
    public static BarangElektronik[] daftar = new BarangElektronik[100];
    public static int jumlahBarang = 0;
    public static Scanner sc = new Scanner(System.in);

    public static int cariIndexById(int id) {
        for (int i = 0; i < jumlahBarang; i++) {
            if (daftar[i].getId() == id)
                return i;
        }
        return -1;
    }

    public static void tambahData() {
        System.out.print("Masukkan ID: ");
        int id = sc.nextInt();
        sc.nextLine();
        if (cariIndexById(id) != -1) {
            System.out.println("ID sudah digunakan.\n");
            return;
        }
        System.out.print("Masukkan Nama: ");
        String nama = sc.nextLine();
        System.out.print("Masukkan Harga: ");
        int harga = sc.nextInt();
        System.out.print("Masukkan Stok: ");
        int stok = sc.nextInt();

        daftar[jumlahBarang] = new BarangElektronik(id, nama, harga, stok);
        jumlahBarang++;
        System.out.println("Data berhasil ditambahkan!\n");
    }

    public static void tampilkanData() {
        System.out.println("\n=== Daftar Barang Elektronik ===");
        for (int i = 0; i < jumlahBarang; i++) {
            System.out.println("ID: " + daftar[i].getId() +
                    ", Nama: " + daftar[i].getNama() +
                    ", Harga: " + daftar[i].getHarga() +
                    ", Stok: " + daftar[i].getStok());
        }
        System.out.println();
    }

    public static void updateData() {
        System.out.print("Masukkan ID yang ingin diupdate: ");
        int id = sc.nextInt();
        sc.nextLine();
        int idx = cariIndexById(id);
        if (idx != -1) {
            System.out.print("Masukkan Nama baru: ");
            String namaBaru = sc.nextLine();
            System.out.print("Masukkan Harga baru: ");
            int hargaBaru = sc.nextInt();
            System.out.print("Masukkan Stok baru: ");
            int stokBaru = sc.nextInt();
            daftar[idx].setNama(namaBaru);
            daftar[idx].setHarga(hargaBaru);
            daftar[idx].setStok(stokBaru);
            System.out.println("Data berhasil diupdate!\n");
        } else {
            System.out.println("Barang tidak ditemukan.\n");
        }
    }

    public static void hapusData() {
        System.out.print("Masukkan ID yang ingin dihapus: ");
        int id = sc.nextInt();
        int idx = cariIndexById(id);
        if (idx != -1) {
            for (int i = idx; i < jumlahBarang - 1; i++) {
                daftar[i] = daftar[i + 1];
            }
            jumlahBarang--;
            System.out.println("Data berhasil dihapus.\n");
        } else {
            System.out.println("Barang tidak ditemukan.\n");
        }
    }

    public static void cariData() {
        System.out.print("Masukkan ID yang ingin dicari: ");
        int id = sc.nextInt();
        int idx = cariIndexById(id);
        if (idx != -1) {
            System.out.println("Ditemukan: ID " + daftar[idx].getId()
                    + ", Nama: " + daftar[idx].getNama()
                    + ", Harga: " + daftar[idx].getHarga()
                    + ", Stok: " + daftar[idx].getStok() + "\n");
        } else {
            System.out.println("Barang tidak ditemukan.\n");
        }
    }
}

File: python/barang_elektronik.py
class BarangElektronik:
    def __init__(self, id=0, nama="", harga=0, stok=0):
        self.__id = id
        self.__nama = nama
        self.__harga = harga
        self.__stok = stok

    # getter dan setter
    def set_id(self, id): self.__id = id
    def get_id(self): return self.__id

    def set_nama(self, nama): self.__nama = nama
    def get_nama(self): return self.__nama

    def set_harga(self, harga): self.__harga = harga
    def get_harga(self): return self.__harga

    def set_stok(self, stok): self.__stok = stok
    def get_stok(self): return self.__stok


# CRUD
daftar = []
# kita simpan list BarangElektronik di sini


def cari_index_by_id(id):
    for i, brg in enumerate(daftar):
        if brg.get_id() == id:
            return i
    return -1


def tambah_data():
    try:
        id = int(input("Masukkan ID: "))
    except ValueError:
        print("ID harus angka")
        return
    if cari_index_by_id(id) != -1:
        print("ID sudah digunakan.\n")
        return

    nama = input("Masukkan Nama: ")
    try:
        harga = int(input("Masukkan Harga: "))
        stok = int(input("Masukkan Stok: "))
    except ValueError:
        print("Harga/Stok harus angka")
        return

    daftar.append(BarangElektronik(id, nama, harga, stok))
    print("Data berhasil ditambahkan!\n")


def tampilkan_data():
    print("\n=== Daftar Barang Elektronik ===")
    for brg in daftar:
        print(f"ID: {brg.get_id()}, Nama: {brg.get_nama()}, Harga: {brg.get_harga()}, Stok: {brg.get_stok()}")
    print()


def update_data():
    try:
        id = int(input("Masukkan ID yang ingin diupdate: "))
    except ValueError:
        print("ID harus angka")
        return
    idx = cari_index_by_id(id)
    if idx != -1:
        nama_baru = input("Masukkan Nama baru: ")
        try:
            harga_baru = int(input("Masukkan Harga baru: "))
            stok_baru = int(input("Masukkan Stok baru: "))
        except ValueError:
            print("Harga/Stok harus angka")
            return
        daftar[idx].set_nama(nama_baru)
        daftar[idx].set_harga(harga_baru)
        daftar[idx].set_stok(stok_baru)
        print("Data berhasil diupdate!\n")
    else:
        print("Barang tidak ditemukan.\n")


def hapus_data():
    try:
        id = int(input("Masukkan ID yang ingin dihapus: "))
    except ValueError:
        print("ID harus angka")
        return
    idx = cari_index_by_id(id)
    if idx != -1:
        daftar.pop(idx)
        print("Data berhasil dihapus.\n")
    else:
        print("Barang tidak ditemukan.\n")


def cari_data():
    try:
        id = int(input("Masukkan ID yang ingin dicari: "))
    except ValueError:
        print("ID harus angka")
        return
    idx = cari_index_by_id(id)
    if idx != -1:
        brg = daftar[idx]
        print(f"Ditemukan: ID {brg.get_id()}, Nama: {brg.get_nama()}, Harga: {brg.get_harga()}, Stok: {brg.get_stok()}\n")
    else:
        print("Barang tidak ditemukan.\n")
